src/metabase/driver/h2.clj

```diff
@@ -467,7 +467,8 @@
 (defmethod sql-jdbc.conn/connection-details->spec :h2
   [_ details]
   {:pre [(map? details)]}
-  (mdb.spec/spec :h2 (update details :db connection-string-set-safe-options)))
+  (mdb.spec/spec :h2 (cond-> details
+                       (string? (:db details)) (update :db connection-string-set-safe-options))))
```

Comment:
Actual fix here. When we create the spec from the *application-db* we don't get a connection-string or db details. So the setting-conn-string-options step isn't needed.